server/src/controllers/game_controller.rs

```diff
@@ -0,0 +1,93 @@
+use crate::models::game_model::Game;
+use actix_web::{guard, http, web, HttpRequest, HttpResponse};
+
+pub fn routes(cfg: &mut web::ServiceConfig) {
+    cfg.service(
+        web::scope("/games")
+            .route(
+                "",
+                web::post()
+                    .guard(guard::Header("content-type", "application/json"))
+                    .to(create),
+            )
+            .route(
+                "/{id}",
+                web::put()
+                    .guard(guard::Header("content-type", "application/json"))
+                    .to(edit),
+            )
+            .route("/{id}", web::get().to(get_one))
+            .route("/{id}", web::delete().to(delete)),
+    );
+}
+
+fn create(_req: HttpRequest, new_game: web::Json<Game>) -> HttpResponse {
+    let mut game = Game {
+        id: 0,
+        start_date: new_game.start_date,
+        is_ended: new_game.is_ended,
+        score: new_game.score,
+        title: new_game.title.to_string(),
+        tracks_number: new_game.tracks_number,
+        is_public: new_game.is_public,
+        mode: new_game.mode,
+    };
+    match game.create() {
+        Err(mut e) => e.send(),
+        _ => HttpResponse::Ok().json(game),
```

Comment:
Maybe it's better if we resurface the id of the created game from the create method of the game model and return it here in the game json. Otherwise, we always return `id: 0`, and the front end might use the id returned here.

Reply:
Are you sure we always return `id: 0`? This is what I get:
```
curl -d "@server/data.json" -H "Content-Type: application/json" -X POST http://localhost:5000/games
{"id":4,"startDate":1578922232856,"isEnded":false,"score":42,"title":"coucou","tracksNumber":21,"isPublic":false,"mode":1}
```

(`data.json` is:
```
{
    "startDate": 1578922232856,
    "isEnded": false,
    "score": 42,
    "title": "coucou",
    "tracksNumber": 21,
    "isPublic": false,
    "mode": 1
}
```
)

The `create` method of the model takes care of updating the `id` but I guess this behavior needs a comment.